src/interface/swarm_device_context.hpp

```diff
@@ -87,16 +87,18 @@ class SwarmDeviceContext {
   KOKKOS_INLINE_FUNCTION
   void Xtoijk(const Real &x, const Real &y, const Real &z, int &i, int &j, int &k) const {
     i = static_cast<int>(
-            std::floor((x - x_min_) / coords_.Dx(CoordinateDirection::X1DIR))) +
+            std::floor((x - x_min_) / coords_.CellWidthFA(CoordinateDirection::X1DIR))) +
         ib_s_;
-    j = (ndim_ > 1) ? static_cast<int>(std::floor(
-                          (y - y_min_) / coords_.Dx(CoordinateDirection::X2DIR))) +
-                          jb_s_
-                    : jb_s_;
-    k = (ndim_ > 2) ? static_cast<int>(std::floor(
-                          (z - z_min_) / coords_.Dx(CoordinateDirection::X3DIR))) +
-                          kb_s_
-                    : kb_s_;
+    j = (ndim_ > 1)
+            ? static_cast<int>(std::floor(
+                  (y - y_min_) / coords_.CellWidthFA(CoordinateDirection::X2DIR))) +
+                  jb_s_
+            : jb_s_;
+    k = (ndim_ > 2)
+            ? static_cast<int>(std::floor(
+                  (z - z_min_) / coords_.CellWidthFA(CoordinateDirection::X3DIR))) +
+                  kb_s_
+            : kb_s_;
```

Comment:
I've noticed lots of places where you've made this change: Dx -> CellWidthFA. I think the better replacement rule would be `Dx -> Dxc<dir>` as that's probably the intent in all these places and it'll be the faster option down the road. For examples it also offers more concise and parthenonic code.

Reply:
If I recall correctly, `Dx` was a run time determined function inherited from Athena++ and thus became `CellWidthFA` from the Athena++ comments on its behavior vs. `dx1v`. I think it might make a difference for some coordinate systems. Everywhere `Dx` was used in Parthenon, however, would be better served by `Dxc<dir>` so I'll make that change